Documentation/EditorConfiguration/VSCodeDevContainer.md

```diff
@@ -0,0 +1,17 @@
+# Visual Studio Code Dev Container
+
+Visual Studio Code has support for containerized development environments, the Ladybird code base has such a Dev Container included to provided a consistent and easily reproducible development environment.
+
+Using the Dev Container requires the installation of the [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension.
+
+After installation the extension will detect the Ladybird Dev Container and provide the option to build it. Start the container build and wait for it to complete, the building process takes quite some time, view the log to monitor progress.
+
+The Dev Container already includes the recommended extensions and settings, as laid out in the [Visual Studio Code Project Configuration](VSCodeConfiguration.md), but it is still a good read for further details.
+
+## Usage
+Run `./Meta/ladybird.py run ladybird` at least once to kick off downloading and building vcpkg dependencies.
+
+On execution, the Ladybird processes run in the container without a browser window. It can be interacted with by connecting to the container using VNC. This can be done from a browser by accessing http://localhost:6080, or from a VNC application by accessing `localhost:5901`. Apps to use are [noVNC](https://novnc.com) in the former case, and [TigerVNC](https://tigervnc.org) in the latter case, or any alternative already available on the system.
```

Comment:
I've been wondering about this bit. On my ubuntu system, VsCode automagically hooks up a wayland socket, and running Ladybird directly from a terminal inside the devcontainer (ubuntu or fedora) "just works" to open windows on my host system.

Does it truly not work on yours?

I also tried running the linux containers from macOS and ran into problem after problem, figuring it wasn't worth the hassle.